File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        Queue<int[]> q = new LinkedList<>();
        // 프로세스의 우선순위와 인덱스를 큐에 추가
        for (int i = 0; i < priorities.length; i++) {
            q.add(new int[] {priorities[i], i});
        }
        
        while(!q.isEmpty()){
            int[] now = q.poll();
            boolean isTrue = false;
            
            for(int[] i: q){
                //현재보다 더 높은 우선순위가 있다면
                if(i[0] > now[0]){
                    isTrue = true;
                    break;
                }
            }
            
            //현재보다 더 높은 우선순위 있는 경우
            if(isTrue){
                q.add(now);
            }
            else{
                //현재 프로세스 실행
                answer++;
                //실행한 프로세스가 찾던 프로세스인 경우 반환
                if(location == now[1]){
                    return answer;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    q = deque()
    
    for i in range(len(priorities)):
        if i == location:
            q.append([priorities[i], 1])
        else:
            q.append([priorities[i], 0])
    
    while q:
        p, find = q.popleft()
        
        # 만약 q에 p보다 우선순위 높은게 있는지 확인
        has_higher = False
        
        for other_p, _ in q:
            if other_p > p:
                has_higher = True
                break
                # q.append([p,find])
        if has_higher:
            q.append([p,find])
        else:
            answer += 1
            if find == 1:
                return answer
    
    return answer